viewer/webpack.config.js

```diff
@@ -31,6 +31,7 @@ module.exports = env => {
     mode: development ? 'development' : 'production',
     entry: {
       index: './src/index.ts',
+      tools: './src/tools/index.ts',
```

Comment:
```js
import { foo } from '@cognite/reveal/tools'
```
Should we have some discussion about this thing? I don't understand the benefits of this. 

These tools will be just a subset of higher-level APIs for the reveal, I'm not sure there is a point to split them into a different package of any kind.

You're basically creating a package inside a package but without proper versioning (versioning is directly bound to reveal, so it's not really a different package or something). It is ok for experimental since nobody really needs it except us, but it feels dubious to me having that for a public API.

A simple alternative for the sake of having lots of tools with nice autocomplete and types – you might create some factory, e.g.

```js
import { RevealTools, Cognite3DViewer } from '@cognite/reveal'

const overlays = RevealTools.create('htmlOverlay', { viewer }); 
// ...
overlays.add(createOverlay(), pos);
```

I would also ask @j-bjorne and @christjt to give some input on that.

Reply:
How do `rxjs` do this? Here we import operators through `rxjs/operators`, other classes through `rxjs`.

If we are to ditch tools, I think we only will go for `import { HtmlOverlayTool } from '@cognite/reveal'` for now and see if we want to add a factory later.

Reply:
rxjs terrifies me as well 😂  

but yeah, I agree some people do this and it is an option, just trying to understand real pros and cons of that approach.

What's for 

> import { HtmlOverlayTool } from '@cognite/reveal'

I don't agree, because we know upfront that we'll have more tools like this, and you can't just add the factory later and remove that possibility of import things (because of breaking changes). And keep 2 ways of doing things probably is not something you want.

Reply:
@larsmoa my biggest concern about that subentry was code duplication between entries, so I built viewer and examples app, then explored with the source-map-explorer tool. I can't see any code duplication for now between entries, all looks fine. 

The only issue I've seen is that types aren't picked up by IDE. I can see viewer build output is `/dist/tools/index.d.ts` + `/dist/tools.js` so if you want to go with sub-entry, it's probably better to move index ts out of the `/src/tools` to have generated js and its types aligned.

![image](https://user-images.githubusercontent.com/4989157/106459632-6fe1bd80-6492-11eb-9c77-ca1d9de3d9b6.png)